From 9645a3641a6d8adca389c14893a102062fe63c41 Mon Sep 17 00:00:00 2001
From: MerCuJerry <mercujerry@gmail.com>
Date: Wed, 19 Nov 2025 10:34:28 +0800
Subject: [PATCH] fix: backport selinux inode

---
 security/selinux/include/objsec.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/security/selinux/include/objsec.h b/security/selinux/include/objsec.h
index c43cd13d8..bf095e8fc 100644
--- a/security/selinux/include/objsec.h
+++ b/security/selinux/include/objsec.h
@@ -142,4 +142,9 @@ struct perf_event_security_struct {
 	u32 sid;  /* SID of perf_event obj creator */
 };
 
+static inline struct inode_security_struct *selinux_inode(const struct inode *inode)
+{
+	return inode->i_security;
+}
+
 #endif /* _SELINUX_OBJSEC_H_ */
-- 
2.52.0

