From 8f3484b4adf8bf6e9bec208586efb1a66000e2e4 Mon Sep 17 00:00:00 2001
From: MerCuJerry <mercujerry@gmail.com>
Date: Sun, 2 Nov 2025 14:26:26 +0800
Subject: [PATCH] fix build error

---
 fs/namespace.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/namespace.c b/fs/namespace.c
index 774419cb6..b61b2ee9d 100644
--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -380,6 +380,7 @@ static struct mount *alloc_vfsmnt(const char *name)
 	if (mnt) {
 		int err;
 
+		err = mnt_alloc_id(mnt);
 		if (err)
 			goto out_free_cache;
 
-- 
2.51.2

