From 10948888e383f33be3d1d6fdf9213bad18f922b5 Mon Sep 17 00:00:00 2001
From: MerCuJerry <mercujerry@gmail.com>
Date: Mon, 17 Nov 2025 12:40:13 +0800
Subject: [PATCH] seccomp

diff --git a/include/linux/seccomp.h b/include/linux/seccomp.h
index a9d5c52de..4adb3b100 100644
--- a/include/linux/seccomp.h
+++ b/include/linux/seccomp.h
@@ -11,6 +11,7 @@
 #ifdef CONFIG_SECCOMP
 
 #include <linux/thread_info.h>
+#include <linux/atomic.h>
 #include <asm/seccomp.h>
 
 struct seccomp_filter;
@@ -27,6 +28,7 @@ struct seccomp_filter;
  */
 struct seccomp {
 	int mode;
+	atomic_t filter_count;
 	struct seccomp_filter *filter;
 };

