From fe1df5edd00384c9cc508082aab54c07d946468b Mon Sep 17 00:00:00 2001
From: MerCuJerry <mercujerry@gmail.com>
Date: Tue, 28 Oct 2025 23:43:18 +0800
Subject: [PATCH] fix build error

---
 fs/namespace.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/fs/namespace.c b/fs/namespace.c
index 9586a5fcc..b1883dbc9 100644
--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -380,6 +380,7 @@ static struct mount *alloc_vfsmnt(const char *name)
 	if (mnt) {
 		int err;
 
+		err = mnt_alloc_id(mnt);
 		if (err)
 			goto out_free_cache;
 
@@ -1207,10 +1208,6 @@ vfs_kern_mount(struct file_system_type *type, int flags, const char *name, void
 {
 	struct mount *mnt;
 	struct dentry *root;
-#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
-	// Make sure mnt->mnt.susfs_mnt_id_backup is initialized every time.
-	mnt->mnt.susfs_mnt_id_backup = 0;
-#endif
 	
 	if (!type)
 		return ERR_PTR(-ENODEV);
-- 
2.51.2

